### DISCUSSION TITLE
Rerender node on actionCtx update

### DISCUSSION THREAD

**Discussion by emcfarlane** [Milkdown/milkdown#612] [Q&A] [answered]
I'm trying to controll a class on a node from an action. I have an `toggleState` based on the "Inject Slices" example here: https://milkdown.dev/plugin-factories

```typescript
const toggleState = createSlice(false, 'toggle');
const toggle = createNode<string>(() => {
    const id = 'toggle';

    return {
        id,
        schema: (ctx) => {
            const myState = ctx.get(toggleState);
            toDOM: (node) => ["div", {class: /* Use state here */ }, 0],
        },
        // ...
    };
}, [toggleState]);
```

It works on new renders but not when updated. Trying to force the update with the `forceUpdate()` util doesn't seem to rerender the node.

Tried force update:
```
  useEffect(() => {
    if (!editorReady) return;
    const instance = editor.getInstance();
    if (!instance) return;
    instance.action((ctx) => ctx.set(toggleState, value));
    instance.action(forceUpdate());
  }, [editorReady, value]);
```

How can you accomplish this? I think I'm not understanding how state changes are handled.

**Comment by emcfarlane** [comment]
Potentially issue with my react `useEditor` have added `useEditor(..., [value])`. Works to rerender.

**Comment by Saul-Mirone** [comment, marked as answer]
`toDOM` is only used to create the node. If you want to update it. You need to set something as [attr](https://prosemirror.net/docs/ref/#model.Node.attrs) and use [`tr.setNodeMarkup`](https://prosemirror.net/docs/ref/#transform.Transform.setNodeMarkup) to update it.
